activerecord/lib/active_record/future_result.rb

```diff
@@ -43,7 +45,7 @@ def execute_or_skip
         return unless @mutex.try_lock
         begin
           if pending?
-            @event_buffer = @instrumenter.buffer
+            @event_buffer = @instrumenter.buffer({ future_result: self })
```

Comment:
Should we make an `@instrumenter.async_buffer` method that allocates a subclass of `Buffer`?  Then we can eliminate the conditional statement in `instrument`.  Or is that overkill?

Reply:
Sounds a bit overkill to me, but I don't mind if you feel strongly about it.